app/src/components/adminConsole/SnappableManager.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState, useRef, useCallback } from "react";
+import GetSnappables from "../../firebase/users/GetSnappables";
+import { snappablesToCsvDownload, readFileAndUpload } from "./csvManager";
+
+const SnappableManager = () => {
+    const [status, setStatus] = useState({ status: "idle" });
+    const fileRef = useRef(null);
+
+    const downloadSnappables = useCallback(() => {
+        setStatus({ status: "loading" });
+        GetSnappables()
+            .then((snappables) => {
+                snappablesToCsvDownload(snappables);
+                setStatus({ status: "idle" });
+            })
+            .catch(() => {
+                setStatus({
+                    status: "error",
+                    error: "There was an error loading snappable people.",
+                });
+            });
```

Comment:
I generally find the `async/await` is easier to read and prefer it over promises.